### PR TITLE
feat (api): fields added to /agents/:agentId/competitions and ensure sorting

### PR DESCRIPTION
**Pull request by mzkrasner**
# Summary

### Issue Overview
GitHub Issue #630 identified that the `/agents/:agentId/competitions` API endpoint was missing critical fields needed by the UI's table sorting component. 

### Required Fields
The UI needed the following agent-specific metrics for each competition:
- `portfolioValue`: Agent's current portfolio value in the competition
- `pnl`: Agent's profit/loss amount and percentage  
- `totalTrades`: Number of trades made by the agent in that competition
- `bestPlacement`: Agent's rank and total participants in the competition

### Solution Implemented
Following the existing `attachAgentMetrics` pattern, we implemented a per-competition metrics enhancement:

**Repository Layer:**
- Added `countAgentTradesInCompetition()` for competition-specific trade counting
- Added `getAgentCompetitionRanking()` for accurate ranking calculations
- Extended sorting support to include computed fields

**Service Layer:**
- Added `attachCompetitionMetrics()` method to enhance competitions with agent-specific data
- Modified `getCompetitionsForAgent()` to apply metrics to each competition
- Implemented proper error handling with sensible defaults

**API Enhancement:**
- Enhanced response includes all required fields with proper TypeScript types
- Full sorting support for new computed fields (`portfolioValue`, `pnl`, `totalTrades`, `rank`)
- Maintained backward compatibility with existing clients

### Testing & Quality
- **37/37 E2E tests passing** covering functionality, edge cases, and multi-agent scenarios
- Comprehensive test coverage including cross-competition ranking accuracy
- Updated OpenAPI documentation with field descriptions and examples

### CODE REVIEW

File: apps/api/src/database/repositories/competition-repository.ts [hunk 1]
```diff
@@ -429,6 +429,53 @@ export async function getAgentPortfolioSnapshots(
...
+    );
+
+    // Find the agent's position (1-based ranking)
+    const agentIndex = sortedSnapshots.findIndex(
+      (snapshot) => snapshot.agentId === agentId,
+    );
+
+    // If agent not found in snapshots, return 0 rank
+    if (agentIndex === -1) {
+      return { rank: 0, totalAgents: sortedSnapshots.length };
```
Comment (dtbuchholz) [inline apps/api/src/database/repositories/competition-repository.ts]: the "rank 0" is problematic because a typical query is: "give my all agents and sort by rank". rank `0` would come before rank `1` and muddy the results.

~~so, we should just make it nullable/undefined (whichever makes it easier wrt our return types). the client can define handling the `undefined / totalAgents` logic.~~

edit: actually, we should just make `bestPlacement` nullable/optional, so the `rank` & `totalAgents` is only supplied if both exist. that way, it's easier to handle on the frontend for rendering: `bestPlacement ? "${rank} / ${totalAgents}" : "N/A"`

File: apps/api/src/services/agent-manager.service.ts [hunk 2]
```diff
@@ -832,13 +842,42 @@ export class AgentManager {
         params,
       );
 
-      // Get agents from repository
+      // Get competitions from repository
       const results = await findAgentCompetitions(agentId, params);
 
+      // Attach metrics to each competition
```
Comment (dtbuchholz) [inline apps/api/src/services/agent-manager.service.ts]: if we do this:
```ts
const competitions = results.competitions.filter(
  (comp) => comp !== null,
);
const enhancedCompetitions = await Promise.all(     
  results.competitions.map(...)
  // ...
)
```
then the `null` checks via `if (!competition) ...` and `validCompetitions` aren't needed.

File: apps/api/src/services/agent-manager.service.ts [hunk 3]
```diff
@@ -1030,6 +1069,123 @@ export class AgentManager {
...
+    }
+  }
+
+  /**
+   * Sort competitions by computed fields
+   * @param competitions Array of competitions with metrics
+   * @param sortString Sort string (e.g., "portfolioValue", "-pnl")
+   * @returns Sorted competitions array
+   */
+  private sortCompetitionsByComputedField(
```
Comment (dtbuchholz) [inline apps/api/src/services/agent-manager.service.ts:1149]: note to self: revisit this logic in https://github.com/recallnet/js-recall/issues/620 

(what you have is good / necessary. we just need to figure out a solution for "load _all_ results in the db query because the post-processing needs to sort the fields, too")

File: apps/api/src/database/repositories/competition-repository.ts [hunk 4]
```diff
@@ -429,6 +429,53 @@ export async function getAgentPortfolioSnapshots(
...
+export async function getAgentCompetitionRanking(
+  agentId: string,
+  competitionId: string,
+): Promise<{ rank: number; totalAgents: number }> {
+  try {
+    // Get all latest portfolio snapshots for the competition
+    const snapshots = await getLatestPortfolioSnapshots(competitionId);
+
+    if (snapshots.length === 0) {
+      return { rank: 0, totalAgents: 0 };
```
Comment (dtbuchholz) [inline apps/api/src/database/repositories/competition-repository.ts]: oh and same thing here re: the point about rank zero & totalAgents zero....if there's no `bestPlacement`, then we should return `undefined` because handling "zero rank" can potentially be misleading for the client to handle.

File: apps/api/src/database/repositories/competition-repository.ts [hunk 5]
```diff
@@ -429,6 +429,53 @@ export async function getAgentPortfolioSnapshots(
...
+      rank: agentIndex + 1, // Convert to 1-based ranking
+      totalAgents: sortedSnapshots.length,
+    };
+  } catch (error) {
+    console.error(
+      "[CompetitionRepository] Error in getAgentCompetitionRanking:",
+      error,
+    );
+    // Return default values on error to prevent breaking the flow
+    return { rank: 0, totalAgents: 0 };
```
Comment (dtbuchholz) [inline apps/api/src/database/repositories/competition-repository.ts]: (and here)

**Comment by dtbuchholz** [comment]
@mzkrasner my only qualm is the `bestPlacement` values for edge cases but everything else looks solid.

i noticed a similar query sort problem while fixing the other comps API in #621. i like the repository-level approach you used here where the response tells you "is this a computed sort field?"...i tried a zod-based one at the service layer and it got a bit messy.

File: apps/api/openapi/openapi.json [hunk 6]
```diff
@@ -15,15 +15,15 @@
   },
   "servers": [
     {
-      "url": "https://api.competitions.recall.network",
+      "url": "https://api.competitions.recall.network/staging",
```
Comment (dtbuchholz) [inline apps/api/openapi/openapi.json]: is there a reason these got changed and include the `staging` prefix and define different ports?

Comment (mzkrasner) [reply apps/api/openapi/openapi.json]: that's odd - let me resolve that

Comment (mzkrasner) [reply apps/api/openapi/openapi.json]: Ah - still had staging assigned to `API_PREFIX` in my local .env which was getting picked up by swagger

**Review by dtbuchholz — APPROVED** [review]
